packages/core/src/components/DataTable/index.tsx

```diff
@@ -128,8 +136,30 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     (...args) => JSON.stringify(args),
   );
 
-  componentDidUpdate(prevProps: DataTableProps) {
+  componentDidUpdate(prevProps: DataTableProps, prevState: State) {
+    const { data, filterData } = this.props;
+    const { sortBy, sortDirection, selectedRows } = this.state;
+    const sortedData: IndexedParentRow[] = this.getData(data!, sortBy, sortDirection, selectedRows);
+    const filteredData = filterData!(sortedData);
+    const oldFilteredData = prevProps.filterData!(sortedData);
+
+    if (
+      filteredData.length > 0 &&
+      (filteredData.length !== oldFilteredData.length ||
+        filteredData.some(
+          (x: IndexedParentRow, i: number) =>
+            x.metadata.originalIndex !== oldFilteredData[i].metadata.originalIndex,
+        ))
+    ) {
+      console.log('123');
+      setTimeout(() => {
```

Comment:
Whats the timeout for?

Reply:
I think this was important for a race condition, if it's needed should def. add a comment given how difficult it was to debug.

Reply:
Yeah, basically react-virutalized tries to measure rows and populate the cache, but React is trying to re-render at the same time, and we want to make sure the measurements are done before React renders the new table.

Reply:
@schillerk can you add a comment as discussed?